packages/mermaid/src/rendering-util/rendering-elements/createLabel.js

```diff
@@ -0,0 +1,105 @@
+import { select } from 'd3';
+import { log } from '$root/logger.js';
+import { getConfig } from '$root/diagram-api/diagramAPI.js';
+import common, { evaluate, renderKatex, hasKatex } from '$root/diagrams/common/common.js';
+import { decodeEntities } from '$root/utils.js';
+
+/**
+ * @param dom
+ * @param styleFn
+ */
+function applyStyle(dom, styleFn) {
+  if (styleFn) {
+    dom.attr('style', styleFn);
+  }
+}
+
+/**
+ * @param {any} node
+ * @returns {Promise<SVGForeignObjectElement>} Node
+ */
+async function addHtmlLabel(node) {
+  const fo = select(document.createElementNS('http://www.w3.org/2000/svg', 'foreignObject'));
+  const div = fo.append('xhtml:div');
+
+  let label = node.label;
+  if (node.label && hasKatex(node.label)) {
+    label = await renderKatex(node.label.replace(common.lineBreakRegex, '\n'), getConfig());
+  }
+  const labelClass = node.isNode ? 'nodeLabel' : 'edgeLabel';
+  div.html(
+    '<span class="' +
+      labelClass +
+      '" ' +
+      (node.labelStyle ? 'style="' + node.labelStyle + '"' : '') +
+      '>' +
+      label +
+      '</span>'
+  );
+
+  applyStyle(div, node.labelStyle);
```

Comment:
We're setting `'style="' + node.labelStyle + '"`, and also calling applyStyle, is both required?

Reply:
Yes, we need both. 
First style apply the style to span, and the applyStyle adds it to the div.